ssas/service/main/main.go

```diff
@@ -337,3 +350,38 @@ func closeRows(rows *sql.Rows) {
 func cliTrackingID() string {
 	return fmt.Sprintf("cli-command-%d", time.Now().Unix())
 }
+
+func getXData(clientID, APIkey string) {
+	if APIkey != "" {
+		const prefix = "Basic "
+		// Check that the APIkey provided that has more than just the `Basic` prefix
+		if len(APIkey) < len(prefix) || !strings.EqualFold(APIkey[:len(prefix)], prefix) {
+			panic("must provide a valid API key")
```

Comment:
Instead of panicking here (and the other spots), can we return an error instead? That way the caller can determine what it wants to do (log the information and then return an exit code?)

Reply:
The reason I stuck with using `panic` instead of bubbling up an error is mostly for keeping it consistent with house the other cli commands are built. But I can revision to make it to do that if we find more value in that.

Reply:
I think it's good for us to start return errors instead. In the case of the CLI, we can offer richer error messaging/formatting. And in a broader sense, returning errors allows the caller of the function to use some alternative strategy (retries, log, etc.) which would be more difficult to reason about if we were using panics